### ISSUE TITLE
Add logs to warn when not all 3 files (mod, info and zip) are present 

### ISSUE THREAD

**Issue by arschles** [gomods/athens#1171]
**Is your feature request related to a problem? Please describe.**

The file system storage currently checks for all three files (`.mod`, `.zip`, `.info`) exist and fails any one endpoint if they don't _all_ exist. This can be problematic if, for some reason, 1/3 or 2/3 of the files exist. In this case, Athens will "fall back" to downloading the module from the VCS and _overwriting_ it in its storage.

That's not an optimal solution, but without a larger refactoring, it's the best we can do

**Describe the solution you'd like**

We should clearly log when we run into a situation when a module doesn't have all 3 files present. The log should be a "warning" with detailed information on why this happens and what Athens is doing to fix it (i.e. falling back to the VCS), and what that means (the module might get overwritten)

**Describe alternatives you've considered**

A larger refactor of how files are stored on disk

**Additional context**

This is a follow-up from https://github.com/gomods/athens/pull/1143#issuecomment-474602212.

**Comment by cByst** [comment]
I'd like to work on this one.

**Comment by arschles** [comment]
@cByst please feel free!

**Comment by arschles** [comment]
@cByst do you think you still want to do this one? Either way, no worries and no rush. Just wanted to check in again.

**Comment by sterlingdeng** [comment]
For this issue, would a good place to start looking be `go_get_fetcher.go`?

**Comment by cByst** [comment]
@arschles @sterlingdeng  yeah I'd like to continue working on this one. Sorry about the delay kinda lost track of this one. Based upon what I was looking at a little while ago, I think it would make sense to change checker.go in the fs package. Would you guys agree? 

It looks like currently the Exists func checks the directory just for the number of file in it. If it has 3 files all is good if not it returns false (it always return nil for the error unless there was an error passed from afero). Do you think it makes senses to add this detailed information as the error in the return when one of the files is missing? if len(files) != 3 then return false and an error with the detailed information you described in the issue?

**Comment by cByst** [comment]
Scratch all that above I've been reading through the code base and see how it works now. It appears that when ```go get``` reaches out to the proxy for the first of the 3 files, if that first file doesn't exist all files are replaced from vcs and the file requested is then served. This pattern continues for all 3 files. Since we don't know if either of the other 2 files exist at the point of checking in the handler for each file do you think it makes sense to just log out that file x of the module was not found so we are going to vcs overwriting anything in storage for this version of this module? Or should I add some logic to get the files in the directory and log out which ones existed and which didn't? Not sure that the extra info of what other files didn't exist is all that helpful and it would add a bit of complexity to the code base, but it shouldn't be to bad if someone thinks that info is worth having. @arschles what do you think?

**Comment by adimv-f** [comment]
Is this still relevant?
PR #1280 hasn't been updated since 2019. Do you mind if I work on this?
